examples/helloworld/CMakeLists.txt

```diff
@@ -16,7 +16,16 @@ if(NOT TARGET CycloneDDS-CXX::ddscxx)
   find_package(CycloneDDS-CXX REQUIRED)
 endif()
 
-idlcxx_generate(TARGET helloworlddata FILES HelloWorldData.idl WARNINGS no-implicit-extensibility)
+set(features_list "")
+if(ENABLE_LEGACY)
+  list(APPEND features_list "union-template=boost::variant")
+  list(APPEND features_list "union-include=\\\"boost/variant.hpp\\\"")
+
+  find_package(Boost)
+  include_directories(${Boost_INCLUDE_DIR})
+endif()
+
+idlcxx_generate(TARGET helloworlddata FILES HelloWorldData.idl WARNINGS no-implicit-extensibility FEATURES ${features_list})
```

Comment:
Is this still needed with the changes `src/idlcxx/src/generator.c`?

Ideally the output of IDLC would be completely target independent, but that would mean having to make this decision in the generated output. That should be possible, but perhaps it is not worth the effort for a "legacy" mode!